### PR TITLE
700: add edtf DatatypeIdValue

### PR DESCRIPTION
**Pull request by Lhaaits**

### CODE REVIEW

File: wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/DatatypeIdImpl.java [hunk 1]
```diff
@@ -101,6 +101,11 @@ public class DatatypeIdImpl implements DatatypeIdValue {
 	 * {@link DatatypeIdValue#DT_GEO_SHAPE} in JSON.
 	 */
 	public static final String JSON_DT_GEO_SHAPE = "geo-shape";
+	/**
+	 * String used to refer to the property datatype
+	 * {@link DatatypeIdValue#DT_TIME} in JSON.
```
Comment (wetneb) [inline wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/DatatypeIdImpl.java:106]: It looks like this comment was copied from one (or two?) other places in the code, but this should be adapted to EDTF (this is not the property datatype).

**Review by wetneb — COMMENTED** [review]
Thanks for the PR!

File: wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/DatatypeIdImpl.java [hunk 2]
```diff
@@ -121,37 +126,39 @@ public class DatatypeIdImpl implements DatatypeIdValue {
...
-		case JSON_DT_STRING:
-			return DT_STRING;
-		case JSON_DT_MONOLINGUAL_TEXT:
-			return DT_MONOLINGUAL_TEXT;
-		default:
-			if(!JSON_DATATYPE_PATTERN.matcher(jsonDatatype).matches()) {
-				throw new IllegalArgumentException("Invalid JSON datatype \"" + jsonDatatype + "\"");
-			}
+			case JSON_DT_ITEM:
+				return DT_ITEM;
```
[the remark this answers is not in the document]
Comment (Lhaaits) [reply wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/DatatypeIdImpl.java]: I think my autoformat fixed more than just my changed lines. I have reverted irrelevant whitespace changes

File: wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/DatatypeIdImpl.java (hunk 1 above)
Comment (Lhaaits) [reply wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/DatatypeIdImpl.java]: Ah yeah, good point. I forgot to update the copied javadoc. Sloppy...

File: wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/DatatypeIdImpl.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Lhaaits) [reply wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/DatatypeIdImpl.java]: Turns out it was an IntelliJ setting that would remove trailing whitespaces on save. So I had to disable the setting to be able to revert the whitespace changes

**Review by wetneb — APPROVED** [review]

**Comment by Lhaaits** [comment]
Thanks for the review!

**Pull request merged**